src/components/common/API.jsx

```diff
@@ -1146,6 +1146,41 @@ class API {
     getColors() {
         return this.colors;
     }
+
+    /**
+     * Vendor
+     */
+
+    async getGProfilerLink(
+        clusteringID,
+        clusterID,
+        topNumFeatures,
+        organism,
+        token
+    ) {
+        let query = {
+            loomFilePath: this.getActiveLoom(),
+            clusteringID: clusteringID,
+            clusterID: clusterID,
+            topNumFeatures: topNumFeatures,
+            organism: token !== null ? '' : organism,
+            token: token === null ? '' : token,
```

Comment:
This could just be 
```suggestion
            organism: token !== null ? token : organism,
```
Then you wouldn't need the extra option in the message, and you wouldn't need to do the check on the backend, the `token` gets passed via `organism` for the request anyway